src/metabase/driver/mongo/query_processor.clj

```diff
@@ -390,14 +452,40 @@
        :collection source-table-name
        :mbql?      true})))
 
+;; omf: if value 'is' a ISODate, extract the date/time string and convert it to date object
+(defn- to-date-object [value]
+  (if (and (string? value) (clojure.string/starts-with? value "ISODate"))
+    (let [reg (re-seq #"ISODate\((.*?)\)" value)
+          isodate (get (first reg) 1)]
+          (c/from-long (c/to-long isodate)))
+    value))
+
+;; omf: possibly reduce-recurse contained map objects rying to find ISODate values
+(defn recurse-map [altered-map k v]
+  (if (map? v)
+    (assoc altered-map k (reduce-kv recurse-map {} v))
+    (assoc altered-map k (to-date-object v)))
+)
```

Comment:
This would be much more idiomatically written like

```clojure
(defn recurse-map [altered-map k v]
  (assoc altered-map k (if map? v
                           (reduce-kv recurse-map {} v)
                           (to-date-object v))))
```